src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -2221,14 +2221,7 @@ void MachUEPNode::emit(CodeBuffer& cbuf, PhaseRegAlloc* ra_) const
 {
   // This is the unverified entry point.
   C2_MacroAssembler _masm(&cbuf);
-
-  __ cmp_klass(j_rarg0, rscratch2, rscratch1);
-  Label skip;
-  // TODO
-  // can we avoid this skip and still use a reloc?
-  __ br(Assembler::EQ, skip);
-  __ far_jump(RuntimeAddress(SharedRuntime::get_ic_miss_stub()));
-  __ bind(skip);
+  __ ic_check(CodeEntryAlignment);
```

Comment:
I'm not sure we want to increase the alignement to CodeEntryAlignment here.  I believe C2 already aligns the root block to CodeEntryAlignment.  @theRealAph, what do you think?

Reply:
I can change it to InteriorEntryAlignment which is what the VEP block is aligned to. I think it's still good that the root block is aligned to CodeEntryAlignment though. I heard some HW sheds some tears when their instruction cache lines also contain data that doesn't correspond to any actual instructions. But we can definitely align to InteriorEntryAlignment here.

Reply:
To add to what Erik is saying here, we have notice that some CPU's have L1I and L1D exclusive, a cache-line can just be in of them. So you can end up with it bouncing between them.

Secondly, if an I-fetch gets data not intended to be decoded, the cpu can get upset in some cases.

Reply:
> Secondly, if an I-fetch gets data not intended to be decoded, the cpu can get upset in some cases.

I've heard that in the past, and always wondered, does it matter if the data is "reachable"?  What if it's on the other side of an unconditional branch?

Reply:
> > Secondly, if an I-fetch gets data not intended to be decoded, the cpu can get upset in some cases.
> 
> I've heard that in the past, and always wondered, does it matter if the data is "reachable"? What if it's on the other side of an unconditional branch?

I don't think it's an guarantee. E.g. an uncond branch may be encoded with multiple instruction, so it might be seen as an indirect jump.